source/v5/mqtt5_to_mqtt3_adapter.c

```diff
@@ -327,6 +344,10 @@ static int s_aws_mqtt_client_connection_5_connect(
     const struct aws_mqtt_connection_options *connection_options) {
 
     struct aws_mqtt_client_connection_5_impl *adapter = impl;
+    int result = s_check_adapter_destroy_status(adapter);
```

Comment:
This check is missing for `s_aws_mqtt_5_resubscribe_existing_topics`, `s_aws_mqtt_client_connection_5_acquire` and `s_aws_mqtt_client_connection_5_release`. I thing, that's all but I could something

Reply:
Updated `s_aws_mqtt_5_resubscribe_existing_topics`.

Debatable: I feel we should directly fail the `acquire`/`release` if the adapter is in destroying process... the ASSERT instead of a validation check might be a better option here.

Reply:
Yeah, this situation (trying to do something meaningful with `connection` in callback while it's being destroyed in another thread) should be pretty rare and most probably will indicate that something's wrong with user code. So, ASSERT could actually help to find a possible issue in the code.